topdown/cache/cache.go

```diff
@@ -134,6 +139,7 @@ func (c *cache) unsafeInsert(k ast.Value, v InterQueryCacheValue) (dropped int)
 			dropKey := key.Value.(ast.Value)
 			c.unsafeDelete(dropKey)
 			c.l.Remove(key)
+			c.keyMtx.Delete(dropKey)
```

Comment:
Can this to remove a mutex that is currently locked by an in-progress request, allowing a new concurrent request with the same key to take place?

Reply:
Good catch. I'll remove this.

Reply:
> If you acquire the lock you can safely remove it

Actually that's not true. Another goroutine could be waiting for a lock on the mutex as it's removed. I think you'd have to lock the entire `keyMtx` dictionary as well, to prevent other goroutines from retrieving the mutex you're about to remove